### DISCUSSION TITLE
Request for Insight or Help to Fix (Any Advice Appreciated)

### DISCUSSION THREAD

**Discussion by Stardog2537** [arkypita/LaserGRBL#1060] [Q&A]
Hello,

I'm a new hobbyist, and I'm having some issues with this software and the machine I recently purchased. I don't think it has a brand name, but that's beside the issue. So, the laser works alright, more or less, except for when attempting to use the lasergrbl vector option. When I send an image as vector to the laser, all isolated elements that do not touch the outline get printed over each other at the bottom of the engraving area. Anything that is part of the outline gets engraved as intended, but if there is a cut out area apart from the outline, like for eyes as an example, those would get burned down at the bottom of the print area. It's almost as if there are no Y-axis values being passed to the board. I went into an image editor to create a work around by making a pass through to those areas so they'd not be cut off from the outline. Doing that works, but isn't ideal. A lot of complex images are off limits to me until I get this sorted.

Thanks in advance.

**Comment by sinfocomp** [comment]
convert to curves

**Comment by Stardog2537** [comment]
Hi, 

Thank you. The screen cap for it is here:
![GrblConfig](https://user-images.githubusercontent.com/6774508/85929572-eb87e600-b883-11ea-9413-af7555f9c1ef.jpg)
Interesting you should mention that. I didn't bring it up, but the measurements are off for both axes. The engraving actually is almost half the size of the actual settings when I run a job. For instance, one job I ran came out close to 2.9 inches when the mm settings should've resulted in a job that was almost 6 (maybe 5.8) inches. They do at least appear to be in the proper proportion for both directions.

**Comment by Stardog2537** [comment]
Okay, thanks to your helpful post, I was able to calibrate the stepper motors to the proper measurement. Now, how do I convert to curves?

**Comment by Stardog2537** [comment]
Now it's doing something else. Let me show you please:
![Example](https://user-images.githubusercontent.com/6774508/85931022-0b70d700-b88f-11ea-82ab-87d4e559e8ad.jpg)
When this figure is being run, the marker that shows the laser position actually goes to the correct places in the image as it does each part.

![Example2](https://user-images.githubusercontent.com/6774508/85931035-24798800-b88f-11ea-8679-70809734dc76.jpg)
But when it actually executes at the machine, you can see the mouth part get placed at the bottom of the image. Strangely, this time through, it's placing the sections behind the knee about where they should go. Yesterday and the day before, it was placing those parts at the bottom around where the mouth parts are going. 

The misalignment issue has to do with the stepper motors I'm fairly certain. The outline misalignment is greatly corrected when I slow the speed way down.

**Comment by Stardog2537** [comment]
Is this due to user error or software?

**Comment by StuartB4** [comment]
Which format is the image you loaded, jpg, bmp, svg .....
Can you upload the origanal here for me to have a look at.

**Comment by Stardog2537** [comment]
![LegoDrake](https://user-images.githubusercontent.com/6774508/85958743-beabff80-b965-11ea-9de7-0ee0f1fc0d59.png)
This is the file. I just take it into LaserGRBL and try to adjust settings to vectorize it. 

The second file below is the workaround I made. By connecting the interior spaces to the outline this way, I avoid the issue altogether.
![LegoDrake2](https://user-images.githubusercontent.com/6774508/85958757-eac78080-b965-11ea-9474-e7d53153cf14.png)

**Comment by Stardog2537** [comment]
The workaround is fine for this image, but for more advanced images, it seems to just dump all the interior outlines together at the bottom of the image. I don't have another example handy at the moment.

**Comment by Stardog2537** [comment]
Oh, sorry, it's a PNG.

**Comment by Stardog2537** [comment]
Strangely, watching the tool marker move in the laserGRBL window, it indeed appears to move up to the appropriate spot in the image to engrave the areas it actually puts at the bottom, and then the tool moves to the outline where it should continue from there. I wonder if I should post the g-code also.

**Comment by StuartB4** [comment]
As far as the image goes it looks fine. Do you load the image in to lasergrbl as it is, or did you vectorize it in another app
like inkscape first.
I just did both the images you posted twice each and both burned in the correct place.
No editing or anything, just loaded as they are.  The largest one is only 60 mm high because I don't have a lot of material to work on at the minute.

![IMG_2434](https://user-images.githubusercontent.com/16980026/85961340-ac4fb700-b9a1-11ea-9c3b-8846595b6e68.JPG)

**Comment by Stardog2537** [comment]
I vectorize in LaserGRBL. I just load it in as I shared it with you, and use the in app option.

**Comment by arkypita** [comment]
99% you have too high acceleration set in your grbl configuration.
Too high acceleration causes the motor to lose position during fast movements between different parts of the image

Read this guide: http://lasergrbl.com/wp-content/uploads/2020/05/Grbl-Configuration-ENG.pdf

**Comment by Stardog2537** [comment]
First, thank you (all of you). I appreciate your help. I have reduced the acceleration, as well as the overall speed. Also, this helped me fix the measurement calibration.

I will try it again with the new settings this afternoon/evening and post the results back here. I cannot use my workshop right now.
